metrics/web/test/dashboard/presentation/state/timer_notifier_test.dart

```diff
@@ -0,0 +1,96 @@
+// Use of this source code is governed by the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+import 'package:fake_async/fake_async.dart';
+import 'package:metrics/dashboard/presentation/state/timer_notifier.dart';
+import 'package:test/test.dart';
+
+import '../../../test_utils/matchers.dart';
+
+void main() {
+  group("TimerNotifier", () {
+    TimerNotifier notifier;
+    const duration = Duration(milliseconds: 10);
+
+    setUp(() {
+      notifier = TimerNotifier();
+    });
+
+    test(
+      ".start() throws an ArgumentError if the given duration is null",
+      () {
+        expect(() => notifier.start(null), throwsArgumentError);
+      },
+    );
+
+    test(
+      ".start() throws an ArgumentError if the given duration is negative",
+      () {
+        const negativeDuration = Duration(seconds: -1);
+
+        expect(() => notifier.start(negativeDuration), throwsArgumentError);
+      },
+    );
+
+    test(
+      ".start() starts a timer with the given duration",
```

Comment:
```suggestion
      ".start() starts ticking with the given duration",
```
Does it make sense? Also, I believe we can test that the `start` method actually creates a periodic timer with the given duration using the `FakeAsync` class.